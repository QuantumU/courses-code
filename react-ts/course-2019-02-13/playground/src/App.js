import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Rating from './Rating';
import HooksDemo from './HooksDemo';

class App extends Component {
  constructor() {
    super();
    this.state = {
      now: new Date(),
      imgId: 0,
      todos: [
        { id: 1, title: 'learn React', completed: false },
        { id: 2, title: 'groceries', completed: true },
        { id: 3, title: 'abc', completed: false }
      ],
      inputText: '',
      rating1: 4,
      rating2: 5
    };
    setInterval(() => {
      this.setState({ now: new Date() });
    }, 1000);
  }

  render() {
    return (
      <div>
        Ein Jahr hat {365 * 24} Stunden.
        <br />
        Heute ist der {`${new Date()}`}. <br />
        Uhrzeit: {this.state.now.toLocaleTimeString()}
        <div>
          <button onClick={this.prev10Img}>{'<'}</button>
          <button onClick={this.prevImg}>{'<'}</button>
          <img src={`http://picsum.photos/200?image=${this.state.imgId}`} />
          <button onClick={this.nextImg}>&gt;</button>
          <ul>
            {this.state.todos.map(todo => (
              <li key={todo.id}>{todo.title}</li>
            ))}
          </ul>
        </div>
        <input
          value={this.state.inputText}
          onChange={event => {
            this.setState({ inputText: event.target.value.toLowerCase() });
          }}
        />
        <Rating stars={this.state.rating1} />
        <Rating
          stars={this.state.rating2}
          onStarsChange={this.handleStars2Change}
        />
        <HooksDemo />
      </div>
    );
  }

  handleStars2Change = stars => {
    this.setState({ rating2: stars });
  };

  nextImg = () => {
    this.setState(oldState => ({ imgId: oldState.imgId + 1 }));
    this.setState(oldState => ({ imgId: oldState.imgId + 1 }));
  };

  prevImg = () => {
    this.changeImgId(this.state.imgId - 1);
  };

  prev10Img = () => {
    this.changeImgId(this.state.imgId - 10);
  };

  changeImgId = id => {
    if (id >= 0) {
      this.setState({ imgId: id });
    }
  };
}

export default App;
