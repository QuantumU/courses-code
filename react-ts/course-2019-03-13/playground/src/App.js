import React, { useState } from 'react';
import Diashow from './Diashow';
import Rating from './Rating';
import DocumentTitle from './DocumentTitle';
import DocumentTitleHooks from './DocumentTitleHooks';

const App = () => {
  const [count, setCount] = useState(0);
  const [coin, setCoin] = useState(Math.random() < 0.5 ? 'Kopf' : 'Zahl');

  const [todos, setTodos] = useState([
    { id: 1, title: 'groceries', completed: false },
    { id: 2, title: 'gardening', completed: true },
    { id: 3, title: 'abc', completed: false }
  ]);

  const [inputText, setInputText] = useState('');
  const [rating1, setRating1] = useState(3);

  return (
    <div>
      <DocumentTitleHooks>{`rating: ${rating1}`}</DocumentTitleHooks>
      Ein Jahr hat {365 * 24} Stunden.
      <div>Heute ist der {new Date().toLocaleDateString()}</div>
      <div>{coin}</div>
      <button
        onClick={() => {
          console.log('hello');
        }}
      >
        say hello
      </button>
      <br />
      <button
        onClick={() => {
          setCount(count + 1);
        }}
      >
        {count}
      </button>
      <button
        onClick={() => {
          setCount(0);
        }}
      >
        reset
      </button>
      <Diashow />
      <h1>Todo</h1>
      {todos.map(todoData => (
        <div key={todoData.id}>{todoData.title}</div>
      ))}
      <h1>Inputs</h1>
      <input
        value={inputText}
        onChange={event => {
          // wir übernehmen den geänderten Wert in unseren state
          setInputText(event.target.value);
        }}
      />
      <h1>Rating</h1>
      <Rating
        stars={rating1}
        onStarsChange={stars => {
          setRating1(stars);
        }}
      />
    </div>
  );
};

export default App;
