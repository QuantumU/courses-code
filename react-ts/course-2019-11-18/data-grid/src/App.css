.active-link {
  color: #222222;
  text-decoration: none;
}
