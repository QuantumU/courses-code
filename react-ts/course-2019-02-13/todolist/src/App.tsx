import React, { useReducer } from 'react';
import './App.css';
import TodoList from './TodoList';
import AddTodo from './AddTodo';
import { Route, Link, Switch } from 'react-router-dom';
import { About } from './About';
import TodosContext from './TodosContext';
import todosReducer from './todosReducer';

const initialTodos = [
  { id: 1, title: 'learn React', completed: false },
  { id: 2, title: 'groceries', completed: true },
  { id: 3, title: 'abc', completed: false }
];

const App = () => {
  const [todos, dispatch] = useReducer(todosReducer, initialTodos);

  return (
    <div className="App">
      <TodosContext.Provider
        value={{
          todos: todos,
          onAdd: newTodoTitle => {
            dispatch({ type: 'ADD_TODO', payload: newTodoTitle });
          },
          onDelete: id => {
            dispatch({ type: 'DELETE_TODO', payload: id });
          },
          onToggle: id => {
            dispatch({ type: 'TOGGLE_TODO', payload: id });
          }
        }}
      >
        <h1>Todo</h1>
        <Link to="/">Home</Link> <Link to="/add">Add</Link>{' '}
        <Link to="/about">About</Link>
        <Switch>
          <Route path="/" exact={true} component={TodoList} />
          <Route
            path="/add"
            exact={true}
            render={props => (
              <AddTodo
                onAdd={newTodoTitle => {
                  dispatch({ type: 'ADD_TODO', payload: newTodoTitle });
                }}
              />
            )}
          />
          <Route path="/about" exact={true} component={About} />
          <Route path="/" render={() => <div>not found</div>} />
        </Switch>
      </TodosContext.Provider>
    </div>
  );
};

export default App;
