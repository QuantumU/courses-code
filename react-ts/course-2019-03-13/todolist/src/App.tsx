import React, { Component } from 'react';
import './App.css';
import TodoType from './TodoType';
import TodoList from './TodoList';
import AddTodo from './AddTodo';

type AppProps = {};
type AppState = {
  todos: Array<TodoType>;
  isFetching: boolean;
};

class App extends Component<AppProps, AppState> {
  constructor(props: AppProps) {
    super(props);
    this.state = {
      todos: [
        { id: 1, title: 'groceries', completed: false },
        { id: 2, title: 'gardening', completed: true },
        { id: 3, title: 'abc', completed: false }
      ],
      isFetching: false
    };
  }

  render() {
    return (
      <div className="App">
        <h1>Todo</h1>
        <TodoList todos={this.state.todos} onCompleted={this.handleCompleted} />
        <AddTodo
          onAddTodo={newTodoTitle => {
            this.setState({
              todos: [
                ...this.state.todos,
                {
                  id:
                    this.state.todos.reduce(
                      (id, todo) => Math.max(id, todo.id),
                      0
                    ) + 1,
                  title: newTodoTitle,
                  completed: false
                }
              ]
            });
          }}
        />
        <button onClick={this.fetchTodos}>fetch from Server</button>
        {this.state.isFetching && <span>Fetching...</span>}
      </div>
    );
  }

  handleCompleted = (id: number) => {
    const newTodos = this.state.todos.map(todo => {
      if (todo.id === id) {
        return { ...todo, completed: true };
      }
      return todo;
    });
    this.setState({ todos: newTodos });
  };

  fetchTodos = () => {
    this.setState({ isFetching: true });
    fetch('https://jsonplaceholder.typicode.com/todos')
      .then(response => response.json())
      .then(serverTodos => {
        this.setState({
          todos: serverTodos.map(
            (serverTodo: TodoType & { userId: number }) => {
              const { userId, ...todo } = serverTodo;
              return todo;
            }
          ),
          isFetching: false
        });
      });
  };
}

export default App;
