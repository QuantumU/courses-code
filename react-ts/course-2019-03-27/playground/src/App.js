import React, { Component, useState } from 'react';
import logo from './logo.svg';
import './App.css';
import Counter from './Counter';
import Slideshow from './Slideshow';
import TodoList from './TodoList';
import Rating from './Rating';
import Button from '@material-ui/core/Button';
import DocumentTitle from './DocumentTitle';

class App extends Component {
  getImgUrl = id => 'https://picsum.photos/200?image=' + id.toString();

  constructor() {
    super();
    this.state = {
      inputText1: '',
      rating1: 3
    };
  }

  // getImgUrl(id) {
  //   return "https://picsum.photos/200?image=" + id.toString()
  // }

  handleStarsChange = newStars => {
    this.setState({ rating1: newStars });
  };

  render() {
    return (
      <div>
        <DocumentTitle>{this.state.inputText1 || '...empty...'}</DocumentTitle>
        Hello world
        <br />A year has {365 * 24} hours
        <div>{new Date().toLocaleDateString()}</div>
        <img src={this.getImgUrl(10)} alt="" />
        <br />
        <Button
          onClick={() => {
            console.log('button clicked');
            alert('button clicked');
          }}
          variant="contained"
          color="primary"
        >
          click me
        </Button>
        <Counter />
        <h2>slideshow</h2>
        <Slideshow />
        <h2>todo list</h2>
        <TodoList />
        <h2>inputs</h2>
        <input
          value={this.state.inputText1}
          onChange={changeEvent => {
            // capture the new content in the state
            this.setState({ inputText1: changeEvent.target.value });
          }}
        />
        <h2>Rating</h2>
        <Rating
          stars={this.state.rating1}
          onStarsChange={this.handleStarsChange}
        />
      </div>
    );
  }
}

export default App;
