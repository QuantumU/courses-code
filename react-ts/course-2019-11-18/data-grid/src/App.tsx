import React, { useState, useEffect, useReducer } from "react";
import { Route, NavLink, Switch } from "react-router-dom";

import "./App.css";
import DataGrid from "./DataGrid";
import ThemeContext from "./ThemeContext";

import { ThemeName } from "./types";
import Settings from "./Settings";
import DocumentTitle from "./DocumentTitle";

import countryReducer from "./countryReducer";
import AddCountryForm from "./AddCountryForm";

type PersonRow = [string, string, string];

type CountryRow = [string, string, string, number];

type ApiCountry = {
  alpha2Code: string;
  name: string;
  capital: string;
  population: number;
};

type StateCountry = [string, string, string, number];

const INITIAL_DATA: Array<PersonRow> = [
  ["Adam", "Riese", "Germany"],
  ["Jackie", "Chan", "China"]
];

const COUNTRY_DATA: Array<CountryRow> = [
  ["AT", "Austria", "Vienna", 8],
  ["DE", "Germany", "Berlin", 80]
];

const App: React.FC = () => {
  const [data, setData] = useState(INITIAL_DATA);

  const [countryData, countryDataDispatch] = useReducer(
    countryReducer,
    COUNTRY_DATA
  );

  const [countryApiData, setCountryApiData] = useState<Array<StateCountry>>([]);
  const [themeName, setThemeName] = useState<ThemeName>("light");
  const [shouldReloadCountryApiData, setShoudReloadCountryApiData] = useState(
    true
  );
  useEffect(
    // Funktion, die nach dem Einbinden der Komponente ausgef체hrt wird
    // und jedesmal, wenn sich props oder state 채ndern
    () => {
      if (shouldReloadCountryApiData) {
        fetch("https://restcountries.eu/rest/v2/all")
          .then(response => response.json())
          .then((data: Array<ApiCountry>) => {
            const countries: Array<StateCountry> = data.map(country => [
              country.alpha2Code,
              country.name,
              country.capital,
              country.population
            ]);
            setCountryApiData(countries);
            setShoudReloadCountryApiData(false);
          });
      } else {
        const timeoutId = setTimeout(() => {
          setShoudReloadCountryApiData(true);
        }, 2000);
        // Aufr채umfunktion zur체ckgeben
        return () => {
          clearTimeout(timeoutId);
        };
      }
    },
    [shouldReloadCountryApiData]
  );

  return (
    <ThemeContext.Provider
      value={{ themeName: themeName, setThemeName: setThemeName }}
    >
      <div
        className="App"
        style={{ backgroundColor: themeName === "light" ? "white" : "grey" }}
      >
        <DocumentTitle>{themeName}</DocumentTitle>

        <div>
          <NavLink to="/" exact={true} activeClassName="active-link">
            Home
          </NavLink>{" "}
          <NavLink to="/settings" activeClassName="active-link">
            Settings
          </NavLink>{" "}
          <NavLink to="/people" activeClassName="active-link">
            People
          </NavLink>{" "}
          <NavLink to="/countries" activeClassName="active-link">
            Countries
          </NavLink>{" "}
          <NavLink to="/countries-from-api" activeClassName="active-link">
            Countries (API)
          </NavLink>
        </div>
        <Switch>
          <Route path="/" exact={true} render={() => <h1>Home</h1>} />
          <Route path="/settings" exact={true} component={Settings} />
          <Route
            path="/people"
            exact={true}
            render={() => (
              <>
                <h2>people</h2>
                <DataGrid data={data} />
              </>
            )}
          />
          <Route
            path="/countries"
            exact={true}
            render={() => (
              <>
                <h2>countries</h2>
                <DataGrid
                  data={countryData}
                  colNames={[
                    "code",
                    "name",
                    "capital",
                    "inhabitants (million)"
                  ]}
                />
                <AddCountryForm
                  onAddCountry={countryData => {
                    countryDataDispatch({
                      type: "ADD_COUNTRY",
                      data: countryData
                    });
                  }}
                />
              </>
            )}
          />
          <Route
            path="/countries-from-api"
            exact={true}
            render={() => (
              <>
                <h2>countries from API</h2>
                <div>
                  <button
                    onClick={() => setShoudReloadCountryApiData(true)}
                    disabled={shouldReloadCountryApiData}
                  >
                    Reload
                  </button>
                  {shouldReloadCountryApiData && "LOADING"}
                </div>
                <DataGrid
                  data={countryApiData}
                  colNames={["code", "name", "capital", "inhabitants"]}
                />
              </>
            )}
          />
          <Route path="/" render={() => <h1>Not found</h1>} />
        </Switch>
      </div>
    </ThemeContext.Provider>
  );
};

export default App;
