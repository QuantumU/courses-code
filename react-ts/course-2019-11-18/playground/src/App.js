import React from 'react';
import './App.css';
import Counter from './Counter';
import Counter_ from './Counter_';
import Slideshow from './Slideshow';
import LoginForm from './LoginForm';
import TodoList from './TodoList';
import Rating from './Rating';
import ReviewList from './ReviewList';

function App() {
  const articleName = 'React';

  const hello = () => {
    alert('hello');
  };

  return (
    <div className="App">
      <h1>hello world</h1>
      <div>Ein Jahr hat {365 * 24} Stunden</div>
      <div>Heute ist der {new Date().toLocaleDateString()}.</div>
      <div>
        <a href={'https://en.wikipedia.org/wiki/' + articleName}>
          some article
        </a>
      </div>
      <div>
        <button onClick={hello}>Say Hello</button>
      </div>
      <span>abc</span> <span>def</span>
      <h2>counter</h2>
      <Counter />
      <Counter_ />
      <h2>slideshow</h2>
      <Slideshow />
      <h2>Login</h2>
      <LoginForm />
      <TodoList />
      <h2>Rating</h2>
      <Rating stars={3} />
      <h2>Reviews</h2>
      <ReviewList />
    </div>
  );
}

export default App;
