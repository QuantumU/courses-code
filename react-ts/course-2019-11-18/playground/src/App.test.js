import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import TestRenderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('snapshot', () => {
  const tree = TestRenderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
