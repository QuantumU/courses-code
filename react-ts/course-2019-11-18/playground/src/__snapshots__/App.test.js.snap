// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="App"
>
  <h1>
    hello world
  </h1>
  <div>
    Ein Jahr hat 
    8760
     Stunden
  </div>
  <div>
    Heute ist der 
    2019-11-20
    .
  </div>
  <div>
    <a
      href="https://en.wikipedia.org/wiki/React"
    >
      some article
    </a>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Say Hello
    </button>
  </div>
  <span>
    abc
  </span>
   
  <span>
    def
  </span>
  <h2>
    counter
  </h2>
  <div>
    <button
      className="btn"
      onClick={[Function]}
    >
      0
    </button>
    <button
      className="btn"
      onClick={[Function]}
    >
      Reset
    </button>
  </div>
  <div>
    <button
      className="btn"
      onClick={[Function]}
    >
      0
    </button>
    <button
      className="btn"
      onClick={[Function]}
    >
      Reset
    </button>
  </div>
  <h2>
    slideshow
  </h2>
  <div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        prev
      </span>
    </button>
    <img
      alt="slideshow"
      src="https://picsum.photos/200?image=10"
    />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        next
      </span>
    </button>
  </div>
  <h2>
    Login
  </h2>
  <form>
    <div>
      <input
        onChange={[Function]}
        value=""
      />
    </div>
    <div>
      <input
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div>
      Password strength: 
      0
    </div>
  </form>
  <h1>
    Todo
  </h1>
  <ul
    className="todolist"
  >
    <li
      className="todo"
    >
      TODO
      : 
      groceries
    </li>
    <li
      className="todo completed"
    >
      DONE
      : 
      cooking
    </li>
    <li
      className="todo"
    >
      TODO
      : 
      gardening
    </li>
  </ul>
  <h2>
    Rating
  </h2>
  <div>
    <span
      onClick={[Function]}
    >
      ★
    </span>
    <span
      onClick={[Function]}
    >
      ★
    </span>
    <span
      onClick={[Function]}
    >
      ★
    </span>
    <span
      onClick={[Function]}
    >
      ☆
    </span>
    <span
      onClick={[Function]}
    >
      ☆
    </span>
    <span
      onClick={[Function]}
    >
      ☆
    </span>
  </div>
  <h2>
    Reviews
  </h2>
  <div>
    <div>
      <div>
        <span
          onClick={[Function]}
        >
          ★
        </span>
        <span
          onClick={[Function]}
        >
          ★
        </span>
        <span
          onClick={[Function]}
        >
          ★
        </span>
        <span
          onClick={[Function]}
        >
          ★
        </span>
        <span
          onClick={[Function]}
        >
          ★
        </span>
        <span
          onClick={[Function]}
        >
          ☆
        </span>
      </div>
      <div>
        <span
          onClick={[Function]}
        >
          ★
        </span>
        <span
          onClick={[Function]}
        >
          ★
        </span>
        <span
          onClick={[Function]}
        >
          ★
        </span>
        <span
          onClick={[Function]}
        >
          ☆
        </span>
        <span
          onClick={[Function]}
        >
          ☆
        </span>
        <span
          onClick={[Function]}
        >
          ☆
        </span>
      </div>
    </div>
    <div>
      average: 
      <div>
        <span
          onClick={[Function]}
        >
          ★
        </span>
        <span
          onClick={[Function]}
        >
          ★
        </span>
        <span
          onClick={[Function]}
        >
          ★
        </span>
        <span
          onClick={[Function]}
        >
          ★
        </span>
        <span
          onClick={[Function]}
        >
          ☆
        </span>
        <span
          onClick={[Function]}
        >
          ☆
        </span>
      </div>
    </div>
  </div>
</div>
`;
